src/order/filter/at_top_bot.lean

```diff
@@ -1168,6 +1168,42 @@ map_at_top_eq_of_gc (λb, b * k + (k - 1)) 1
     calc b = (b * k) / k : by rw [nat.mul_div_cancel b hk]
       ... ≤ (b * k + (k - 1)) / k : nat.div_le_div_right $ nat.le_add_right _ _)
 
+/-- A monotone sequence of natural numbers either tends to infinity *or* is eventually constant. -/
+lemma monotone.tendsto_at_top_at_top_or_eventually_const_nat [semilattice_sup ι] [nonempty ι]
+  {f : ι → ℕ} (hf : monotone f) : tendsto f at_top at_top ∨ ∃ x, f =ᶠ[at_top] (λ _, x) :=
```

Comment:
The assumption that iota is nonempty is not necessary here: if it is empty, then the first clause of your disjunction is satisfied.
Also, this lemma and the next one are so similar, that I would even prefer a more general version covering the two of them simultaneously. Can you identify typeclass assumptions for which the proof would go through in both situations?

Reply:
My understanding is that this works on (uniformly) discrete spaces with a total order, but I don't know if that notion exists in Mathlib currently, and if it did I don't know how you would do the induction used in this version of the proof on a generic type. Perhaps there's a better way? I'm a computer scientist, not a mathematician, so I'm learning the topology stuff as I go. Any hints you can give would be immensely helpful.

We have [`discrete_topology`](https://leanprover-community.github.io/mathlib_docs/topology/order.html#discrete_topology), but that's not enough AFAICT. From Wikipedia,

> The topology underlying a metric space can be discrete, without the metric being uniformly discrete: for example the usual metric on the set {2ⁿ : n ∈ ℕ}.

Reply:
If these lemmas are still relevant, I'll generalize them to avoid `nonempty` as you suggest.